src/legacy/core_plugins/kbn_vislib_vis_types/public/utils/legend_positions.ts

```diff
@@ -17,18 +17,34 @@
  * under the License.
  */
 
-import { uiModules } from 'ui/modules';
-import vislibGridTemplate from './grid.html';
-const module = uiModules.get('kibana');
+import { i18n } from '@kbn/i18n';
+import { LegendPositions } from '../types';
 
-module.directive('vislibGrid', function () {
-  return {
-    restrict: 'E',
-    template: vislibGridTemplate,
-    replace: true,
-    link: function ($scope) {
+const legendPositions = [
+  {
+    text: i18n.translate('kbnVislibVisTypes.legendPositions.topText', {
+      defaultMessage: 'Top',
+    }),
+    value: LegendPositions.TOP,
+  },
+  {
+    text: i18n.translate('kbnVislibVisTypes.legendPositions.leftText', {
+      defaultMessage: 'Left',
+    }),
+    value: LegendPositions.LEFT,
+  },
+  {
+    text: i18n.translate('kbnVislibVisTypes.legendPositions.rightText', {
+      defaultMessage: 'Right',
+    }),
+    value: LegendPositions.RIGHT,
+  },
+  {
+    text: i18n.translate('kbnVislibVisTypes.legendPositions.bottomText', {
+      defaultMessage: 'Bottom',
+    }),
+    value: LegendPositions.BOTTOM,
+  },
+];
 
-      $scope.isGridOpen = true;
-    }
-  };
-});
+export { legendPositions };
```

Comment:
Let's add `legendPositions` into `editorConfig.collections` to be in consistency with other configs

Reply:
Done.